pkg/csi/service/wcp/controller.go

```diff
@@ -376,8 +378,14 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 		case common.AttributeStoragePool:
 			storagePool = req.Parameters[paramName]
 		case common.AttributeStorageTopologyType:
-			// TODO: TKGS-HA : Add validation
+			// TKGS-HA: validate storageTopologyType.
 			storageTopologyType = req.Parameters[paramName]
+			val := strings.ToLower(storageTopologyType)
+			if val != "zonal" && val != "crosszonal" {
+				return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCodef(log, codes.InvalidArgument,
```

Comment:
make sure we return final error from here.

Reply:
We always return `InvalidArgumentFault` in such situations. (Checked vanilla code too)